version: '3.7'

services:
  covid19-stats-app:
    # image: aritrarudra/covid19-stats-app:0.0.1-SNAPSHOT
    # image: aritrarudra/covid19-stats-app:latest
    build:
      # context: covid-app
      dockerfile: Dockerfile
    ports:
      - "80:8080"
    networks:
      - internal-db

  mysqldb:
    image: mysql:8.0.19
    volumes:
      - ./mysql-data:/var/lib/mysql
      # - ./mysql-init-files:/docker-entrypoint-initdb.d
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_DATABASE: appdb
      MYSQL_USER: appuser
      MYSQL_PASSWORD: appuser
    ports:
      - "3306:3306"
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      timeout: 20s
      retries: 10
networks:
  internal-db: